src/components/implementation/no_interface/cFE_booter/osfiles.c

```diff
@@ -17,201 +23,260 @@ int32 OS_FS_Init(void)
-    return 0;
+    return file_lseek(filedes, offset, whence);
 }
 
 /*
  * Removes a file from the file system
 */
 int32 OS_remove(const char *path)
 {
-    PANIC("Read Only Filesystem!");
-    return 0;
+    int32 ret = path_chk_exists(path);
+    if (ret != OS_FS_SUCCESS) return ret;
+    ret = file_remove((char *)path);
+    return ret;
 }
 
 /*
  * Renames a file in the file system
 */
-int32 OS_rename(const char *old_filename, const char *new_filename)
-{
-    PANIC("Read Only Filesystem!");
-    return 0;
+int32 OS_rename(const char *old_filename, const char *new_filename) {
+    if (!old_filename || !new_filename) return OS_FS_ERR_INVALID_POINTER;
+    int32 ret = path_chk_exists(old_filename);
+    if (ret != OS_FS_SUCCESS) return ret;
+    ret = path_chk_isvalid(new_filename);
+    if (ret != OS_FS_SUCCESS) return ret;
+    return file_rename((char *)old_filename, (char *)new_filename);
 }
 
 /*
  * copies a single file from src to dest
 */
 int32 OS_cp(const char *src, const char *dest)
 {
-    PANIC("Read Only Filesystem!");
-    return 0;
+    if (!src || !dest) return OS_FS_ERR_INVALID_POINTER;
+    char *src_path  = (char *)src;
+    char *dest_path = (char *)dest;
+
+    if (strlen(src_path) > OS_MAX_PATH_LEN) return OS_FS_ERR_PATH_TOO_LONG;
+    if (strlen(dest_path) > OS_MAX_PATH_LEN) return OS_FS_ERR_PATH_TOO_LONG;
+    if (strlen(path_to_name(src_path)) > OS_MAX_FILE_NAME) return OS_FS_ERR_NAME_TOO_LONG;
+    if (strlen(path_to_name(dest_path)) > OS_MAX_FILE_NAME) return OS_FS_ERR_NAME_TOO_LONG;
+
+    int32 ret = path_chk_exists(src_path);
+    if (ret != OS_FS_SUCCESS) return ret;
+    ret = path_chk_isvalid(dest_path);
+    if (ret != OS_FS_SUCCESS) return ret;
+
+    return file_cp(src_path, dest_path);
 }
 
 /*
  * moves a single file from src to dest
 */
 int32 OS_mv(const char *src, const char *dest)
 {
-    PANIC("Read Only Filesystem!");
-    return 0;
+    char *src_path  = (char *)src;
+    char *dest_path = (char *)dest;
+
+    if (!src_path || !dest_path) return OS_FS_ERR_INVALID_POINTER;
+    if (strlen(src_path) > OS_MAX_PATH_LEN) return OS_FS_ERR_PATH_TOO_LONG;
+    if (strlen(dest_path) > OS_MAX_PATH_LEN) return OS_FS_ERR_PATH_TOO_LONG;
+    if (strlen(path_to_name(src_path)) > OS_MAX_FILE_NAME) return OS_FS_ERR_NAME_TOO_LONG;
+    if (strlen(path_to_name(dest_path)) > OS_MAX_FILE_NAME) return OS_FS_ERR_NAME_TOO_LONG;
+
+    int32 ret = path_chk_exists(src);
+    if (ret != OS_FS_SUCCESS) return ret;
+    ret = path_chk_isvalid(dest);
+    if (ret != OS_FS_SUCCESS) return ret;
+    ret = file_mv(src_path, dest_path);
+
+    return ret;
 }
 
 /*
  * Copies the info of an open file to the structure
 */
 int32 OS_FDGetInfo(int32 filedes, OS_FDTableEntry *fd_prop)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    if (!fd_prop)                                   return OS_FS_ERR_INVALID_POINTER;
+    if (filedes <= 0 || filedes > MAX_NUM_FILES)    return OS_FS_ERR_INVALID_FD;
```

Comment:
@zacharied we can't leave debug prints in this kind of code. Logging would be great though...